### ISSUE TITLE
PROPFIND with Depth fails on Serve WebDAV 

### ISSUE THREAD

**Issue by alensiljak** [rclone/rclone#7492]
I just ran into this issue after trying PROPFIND with 1.65. 
Technically, this is a follow-up on the `serve webdav` issues. #7444 and #7448. It seems that #6980 and perhaps others might also be related.

It was mentioned in the last comment [here](https://github.com/rclone/rclone/issues/7444#issuecomment-1828943189), that the message is `Request header field depth is not allowed by Access-Control-Allow-Headers in preflight response.`
This effectively prevents listing just the requested folder. The PROPFIND request works but it does not receive the Depth parameter, which is sent through the header. This comment contains the links to the WebDAV headers and the line that needs to be modified.

Additionally ,as @ncw mentioned [here](https://github.com/rclone/rclone/issues/7444#issuecomment-1838426670), this should be done in the webdav server, not the global one.

#### The associated forum post URL from `https://forum.rclone.org`

n/a

#### What is the problem you are having with rclone?

Issuing a PROPFIND request fails if it contains a `depth:1` header.

#### What is your rclone version (output from `rclone version`)

1.65

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)

Win 11, 64-bit

#### Which cloud storage system are you using? (e.g. Google Drive)

local drive

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)

rclone serve webdav . --allow-origin "*"

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)

n/a

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by alensiljak** [comment]
The client I'm using seems to also send the request payload.

```xml
<?xml version="1.0" encoding="utf-8"?>
<D:propfind xmlns:D="DAV:">
  <D:allprop />
</D:propfind>
```

**Comment by alensiljak** [comment]
I suppose the changes should go somewhere here?
https://github.com/rclone/rclone/blob/2f5685b40562cf764da673543245ddb13c049a4a/cmd/serve/webdav/webdav.go#L197

**Comment by alensiljak** [comment]
Another request that fails is the file download (GET)

```
Access to fetch at 'http://localhost:8080/file.txt' from origin 'http://localhost:5000' has been blocked by CORS policy: Request header field translate is not allowed by Access-Control-Allow-Headers in preflight response.
```

**Comment by justgook** [comment]
problem is here - it missing `depth`:
https://github.com/rclone/rclone/blob/eef0b39a2caf03dc8b14cf2eb69d6c210499aa90/lib/http/middleware.go#L192
while next line:
https://github.com/rclone/rclone/blob/eef0b39a2caf03dc8b14cf2eb69d6c210499aa90/lib/http/middleware.go#L193
includes all webDAV related params.

**Comment by alensiljak** [comment]
Is there a problem with #7495?

**Comment by justgook** [comment]
#7495 is not merged and has conflicts that must be resolved. and looking on changes in few yers looks like serve middleware just included all methods and headers, instead of creating dedicated configuration for that.. for me that is just game stopper, that can be resolved with few headers in middleware.. and i just hope that another pull request that is closer to patterns of how app is develop, i just hope that it be more likely be merged than 2y old PR.

**Comment by alensiljak** [comment]
It was implemented in such a way because that is exactly how the maintainer requested it.
But, anyway, good luck.

**Comment by justgook** [comment]
yup - i saw that comment, just it is weird to me that it is still not merged, while there exists just methods that is used only for WebDav, and not Headers, in addition (maybe it just for me looks ugly) “authorization”  header is not capitalized. i just hope that some of PR will be closed, and some merged.
